rust/worker/src/cache/mod.rs

```diff
@@ -0,0 +1,16 @@
+pub mod cache;
+pub mod config;
+use crate::cache::cache::Cache;
+use crate::cache::config::CacheConfig;
+use crate::errors::ChromaError;
+use std::hash::Hash;
+
+pub(crate) async fn from_config<K, V>(
+    config: &CacheConfig,
+) -> Result<Cache<K, V>, Box<dyn ChromaError>>
+where
+    K: Send + Sync + Hash + Eq + 'static,
+    V: Send + Sync + Clone + 'static,
+{
+    Ok(Cache::new(config))
```

Comment:
I think the new() code paths panic if the configuration is invalid, whereas it seems like try_from_config is also implemented. I think the pattern we should follow is to implement Configurable for Cache<K,V> this from_config thing is a bad pattern we should rectify wherever used. Then the callsites on the way down to the concrete type should also use try_from_config so that the error can be propagated up

Reply:
Agree.